src/com/google/ProductionInstrumentationReporter.java

```diff
@@ -0,0 +1,442 @@
+        // If it is, we will average the data, otherwise we will add to the average as if it is 0.
+        for (Map<String, ProfilingData> instrumentationData : allInstrumentationReports) {
+          ProfilingData profilingData = instrumentationData.get(param);
+          runningAverageCounter++;
+          if (profilingData != null) {
+            // if executed, add 1 to executionAverage, and 0 otherwise.
+            averageFrequency += profilingData.frequency;
+            executionAverage++;
+          }
+        }
+
+        ProfilingResult profilingResult = new ProfilingResult(instrumentationMapping, param);
+        // Round the executionAverage to 2 decimal places for simplicity.
+        profilingResult.executed = roundFloat2Decimals(executionAverage / runningAverageCounter);
+        profilingResult.data = new ProfilingData(
+            Math.round((float) averageFrequency / runningAverageCounter));
+
+        if (profilingDataPerFunction.containsKey(functionName)) {
+          profilingDataPerFunction.get(functionName).add(profilingResult);
+        } else {
+          profilingDataPerFunction
+              .put(functionName, new ArrayList<>(Arrays.asList(profilingResult)));
+        }
+
+      }
+
+      ProfilingResultByFile profilingResultByFile = new ProfilingResultByFile();
+      profilingResultByFile.fileName = fileName;
+      profilingResultByFile.profilingDataPerFunction = profilingDataPerFunction;
+
+      result.add(profilingResultByFile);
+
+    }
+
+    return result;
+  }
+
+  /**
+   * Reads all files found in folder and converts the contents of each file to a Map<String,
+   * ProfilingData> data structure where it is a mapping of the unique param value to the encoded
+   * values. The folder contains all the reports sent by the instrumented production code.
+   */
+  private static List<Map<String, ProfilingData>> getAllExecutionResults(File folder)
+      throws IOException {
+    List<Map<String, ProfilingData>> result = new ArrayList<>();
+
+    for (final File fileEntry : folder.listFiles()) {
+      String executionResult = readFile(fileEntry.getAbsolutePath());
+      Type type = new TypeToken<Map<String, ProfilingData>>() {
+      }.getType();
+      Map<String, ProfilingData> executedInstrumentationData = new Gson()
+          .fromJson(executionResult, type);
+
+      result.add(executedInstrumentationData);
+
+    }
+
+    return result;
+  }
+
+  /**
+   * Creates a file with the given fileName (including extension) with the contents of the file
+   * being provided by fileContents.
+   */
+  private static void createFile(String fileName, String fileContents) throws IOException {
+
+    File fold = new File(fileName);
+    fold.delete();
+    File myObj = new File(fileName);
+    myObj.createNewFile();
+
+    FileWriter myWriter = new FileWriter(fileName);
+    myWriter.write(fileContents);
+    myWriter.close();
+  }
+
+  public static void main(String[] args) throws IOException {
```

Comment:
This function will be modified to accept dynamic values through the command line obviously.